src/Streamly/Internal/Data/Stream/IsStream/Transform.hs

```diff
@@ -1039,8 +1041,16 @@ takeWhileAround = undefined -- fromStreamD $ D.takeWhileAround n $ toStreamD m
 -- /Pre-release/
 --
 {-# INLINE takeInterval #-}
-takeInterval ::(MonadIO m, IsStream t, TimeUnit64 d) => d -> t m a -> t m a
-takeInterval d = fromStreamD . D.takeByTime d . toStreamD
+takeInterval ::
+       (MonadAsync m, IsStream t, TimeUnit64 d, Functor (t m))
+    => Double -> d -> t m a -> t m a
+takeInterval g duration = catMaybes . Par.parallelMin timeStream . fmap Just
+
+    where
+
+    duration64 = toRelTime64 duration
+    timeStream =
+        fmap (const Nothing) $ takeWhile (< duration64) (relTimesWith g)
```

Comment:
Can't we implement this in the same way as `dropInterval`? that is by using zipWith?